internal/pkg/cli/task_run.go

```diff
@@ -134,6 +175,172 @@ func (o *runTaskOpts) Ask() error {
 	return nil
 }
 
+// Execute create or update task resources and run the task.
+func (o *runTaskOpts) Execute() error {
+	if o.image == "" && o.dockerfilePath == "" {
+		o.dockerfilePath = defaultDockerfilePath
+	}
+
+	if o.imageTag == "" {
+		o.imageTag = defaultImageTag
+	}
+
+	if err := o.getNetworkConfig(); err != nil {
+		return err
+	}
+
+	cluster, err := o.getCluster()
+	if err != nil {
+		return err
+	}
+
+	if err := o.deployTaskResources(); err != nil {
+		return err
+	}
+
+	//if image is not provided, then we build the image and push to ECR repo
+	if o.image == "" {
+		uri, err := o.pushToECRRepo()
+		if err != nil {
+			return err
+		}
+		o.image = fmt.Sprintf(fmtImageURL, uri, o.imageTag)
+
+		// update image to stack
+		if err := o.updateTaskResources(); err != nil {
+			return err
+		}
+	}
+
+	return o.runTask(cluster)
+}
+
+func (o *runTaskOpts) getCluster() (string, error) {
+	if o.env == config.EnvNameNone {
```

Comment:
I find the `EnvNameNone` a bit confusing, so as a customer do I have to explicitly write "None" to the `-e` flag for it to pick up the default cluster?

Why can't we use the default empty string `""` to say that a user hasn't specified an environment? 
I'd love for us to think if we can simplify this for our customers since the "None" environment doesn't exist in our other commands.

Reply:
No, they don't explicitly specify "None". The first time "None" appears is when we ask for an env name during `Ask`. We list all the environments of the app, and append to the end an option called "None" which would result in the default cluster and subnets if selected. 

We can use "" instead of "None". As a customer the experience would be the same - the "None" would still be a reserved keyword because it's till gonna be there during `Ask`, so that the user can opt to use the default cluster/subnet during `Ask`.

Reply:
Gotcha! that's good to know :) ok then I think it's fine but maybe we can move the const out of the `config` pkg to the `cli` package itself. Since `config` pkg itself doesn't use the constant.

Reply:
Sounds good!!